src/io/mmap/arena.rs

```diff
@@ -64,29 +84,45 @@ impl<'a> Arena<'a> {
         }
 
         for index in 0..v4l2_reqbufs.count {
+            let mut v4l2_planes: Vec<v4l2_plane> = Vec::new();
+            unsafe {
+                v4l2_planes.resize(num_planes as usize, mem::zeroed());
+            }
             let mut v4l2_buf = v4l2_buffer {
                 index,
                 ..self.buffer_desc()
             };
+            if self.buf_type.planar() {
+                v4l2_buf.length = num_planes as u32;
+                v4l2_buf.m.planes = v4l2_planes.as_mut_ptr();
+            }
             unsafe {
                 v4l2::ioctl(
                     self.handle.fd(),
                     v4l2::vidioc::VIDIOC_QUERYBUF,
                     &mut v4l2_buf as *mut _ as *mut std::os::raw::c_void,
                 )?;
 
-                let ptr = v4l2::mmap(
-                    ptr::null_mut(),
-                    v4l2_buf.length as usize,
-                    libc::PROT_READ | libc::PROT_WRITE,
-                    libc::MAP_SHARED,
-                    self.handle.fd(),
-                    v4l2_buf.m.offset as libc::off_t,
-                )?;
+                // each plane has to be mapped separately
+                let mut planes = Vec::new();
+                for plane in &v4l2_planes {
+                    let ptr = v4l2::mmap(
+                        ptr::null_mut(),
+                        plane.length as usize,
+                        libc::PROT_READ | libc::PROT_WRITE,
+                        libc::MAP_SHARED,
+                        self.handle.fd(),
+                        plane.m.mem_offset as libc::off_t,
+                    )?;
+
+                    planes.push(slice::from_raw_parts_mut::<u8>(
+                        ptr as *mut u8, plane.length as usize
+                    ));
```

Comment:
When I tried to use this branch for my application, which use both multi plane and single plane, I received error with single plane device. This fix works with my environment.

```suggestion
                    let length = if !self.buf_type.planar() { v4l2_buf.length as usize } else { plane.length as usize };
                    let ptr = v4l2::mmap(
                        ptr::null_mut(),
                        length,
                        libc::PROT_READ | libc::PROT_WRITE,
                        libc::MAP_SHARED,
                        self.handle.fd(),
                        plane.m.mem_offset as libc::off_t,
                    )?;

                    planes.push(slice::from_raw_parts_mut::<u8>(
                        ptr as *mut u8, length,
                    ));
```